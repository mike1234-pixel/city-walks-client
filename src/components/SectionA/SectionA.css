/** @format */

.section-a {
  width: 100%;
  max-width: 100%;
  height: 75vh;
}

.grid-container-a {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: 100%;
  grid-gap: 30px;
  padding: 5%;
}

@media (max-width: 650px) {
  .grid-container-a {
    grid-template-columns: 100%;
    grid-template-rows: repeat(2, 1fr);
  }
}
@media (max-height: 1400px) and (max-width: 1050px) {
  .section-a {
    height: 120vh;
  }

  .grid-container-a {
    grid-template-columns: 100%;
    grid-template-rows: repeat(2, 45%);
    font-size: 24px;
  }
}

.grid-item-a {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  transition: 1s;
}

.grid-item-a:hover,
.grid-item-a:active {
  background: #fffffe;
  color: #333333;
  cursor: pointer;
}

@media (max-width: 1050px) {
  .grid-item-a {
    font-size: 55%;
  }
}

.grid-item-a-text-box {
  padding: 20px;
  text-align: center;
  background: var(--elegant-color-dark);
  color: white;
}

@media (max-width: 1050px) {
  .grid-item-a {
    padding: 10px;
  }
}

.grid-item-a-img {
  width: 100%;
  height: 100%;
}
