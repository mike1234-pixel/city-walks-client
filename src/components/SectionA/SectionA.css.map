{"version":3,"sourceRoot":"","sources":["SectionA.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA;;;AAGF;EACE;IACE;IACA;IACA;IACA;IACA;;;AAIJ;EACE;IACE;;EAGF;IACE;;;AAIJ;EACE;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;;;AAGF;AAAA;EAEE;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;AAEA;EAGE;;;AAIJ;EACE;EACA","file":"SectionA.css"}