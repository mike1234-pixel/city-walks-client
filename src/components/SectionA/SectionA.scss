/** @format */

.section-a {
  width: 100%;
  max-width: 100%;
}

@media (min-width: 991px) {
  .grid-container-a {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: 100%;
    grid-gap: 30px;
    padding: 30px 0;
  }
}

@media (max-width: 992px) {
  .grid-item-a-text-box {
    margin-bottom: 30px;
  }
  .grid-item-a-image-box {
    margin-bottom: 30px;
  }
}

.grid-item-a {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  transition: all 0.4s;
  box-sizing: border-box;
  border: 2px solid transparent;
}

#grid-item-overlay {
  height: 100%;
  width: 100%;
}

.grid-item-a:hover,
.grid-item-a:active {
  background: transparent;
  color: var(--elegant-color-dark);
  cursor: pointer;
}

.grid-item-a-text-box {
  padding: 20px;
  text-align: left;
  background: var(--elegant-color-dark);
  color: white;
  &:hover,
  &:focus,
  &:active {
    border: 2px dashed grey;
  }
}

.grid-item-a-img {
  width: 100%;
  height: 100%;
}
