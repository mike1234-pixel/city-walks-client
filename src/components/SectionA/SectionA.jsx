import { MDBView } from "mdbreact"
import marked from "marked"
import "./SectionA.css"

const SectionA = (props) => {

    const {content, img, alt} = props

    const createMarkup = (markup) => {
      return {__html: marked(markup)}
    }

  return (
    <div className='section-a grid-container-a'>
        <div className='grid-item-a grid-item-a-text-box'>
            <div dangerouslySetInnerHTML={createMarkup(content)}></div> 
        </div>
        <div className='grid-item-a grid-item-a-image-box'>
          <MDBView hover zoom>
            <img className="grid-item-a-img" src={img} alt={alt}></img>
          </MDBView>
        </div>
    </div>
  );
};

export default SectionA

// content: no more than 570 characters