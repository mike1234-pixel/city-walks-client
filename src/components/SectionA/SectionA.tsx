import React from 'react'
import { MDBView } from "mdbreact"
import marked from "marked"
import "./SectionA.scss"

interface Props {
  content: string;
  img: string;
  alt: string;
}

const SectionA: React.FC<Props> = (props: Props) => {

  const { content, img, alt } = props

  const createMarkup = (markup: string) => {
    return { __html: marked(markup) }
  }

  return (
    <div className='section-a grid-container-a'>
      <div className='grid-item-a grid-item-a-text-box'>
        <div dangerouslySetInnerHTML={createMarkup(content)}></div>
      </div>
      <div className='grid-item-a grid-item-a-image-box'>
        <MDBView hover zoom id="grid-item-overlay" waves>
          <img className="grid-item-a-img" src={img} alt={alt}></img>
        </MDBView>
      </div>
    </div>
  );
};

export default SectionA

// content: no more than 570 characters