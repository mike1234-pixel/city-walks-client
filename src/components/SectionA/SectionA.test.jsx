import React from 'react'
import renderer from 'react-test-renderer'
import SectionA from './SectionA'

// snapshot test
it('Section A renders correctly', () => {  
    
    const content = "mock content"
    const img = "imagefile"
    const alt = "alt attribute"

  const tree = renderer
    .create(
        <SectionA content={content} img={img} alt={alt}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
