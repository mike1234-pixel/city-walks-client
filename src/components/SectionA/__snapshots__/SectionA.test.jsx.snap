// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section A renders correctly 1`] = `
<div
  className="section-a grid-container-a"
>
  <div
    className="grid-item-a grid-item-a-text-box"
  >
    <p>
      mock content
    </p>
  </div>
  <div
    className="grid-item-a grid-item-a-image-box"
  >
    <div
      className="view overlay zoom"
      data-test="view"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <img
        alt="alt attribute"
        className="grid-item-a-img"
        src="imagefile"
      />
    </div>
  </div>
</div>
`;
